src/reverse_complement.rs

```diff
@@ -174,17 +209,19 @@ unsafe impl<T: 'static> Send for Racy<T> {}
 /// The closure `f` is run in parallel with an element of `iter`.
 fn parallel<'a, I, T, F>(iter: I, ref f: F)
         where T: 'static + Send + Sync,
-              I: Iterator<Item=&'a mut [T]>,
-              F: Fn(&mut [T]) + Sync {
-    let jhs = iter.map(|chunk| {
+              I: Iterator<Item=(&'a mut [T], &'a mut [T])>,
+              F: Fn(&mut [T], &mut [T]) + Sync {
+    let jhs = iter.map(|(a, b)| {
```

Comment:
`parallel` is really not idiomatic. It was written when scoped_spawn was discovered unsound. Now, we can write that safely using crossbeam. 

But maybe a rayon approach is more appropriate?